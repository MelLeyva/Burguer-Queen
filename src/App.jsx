import React from 'react';
import LogIn from './Components/Log-in';
import './Styles-scss/App.scss';

function App() {
  return <LogIn />;
}

export default App;
