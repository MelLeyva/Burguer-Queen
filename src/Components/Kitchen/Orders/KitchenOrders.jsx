/* eslint-disable react/prop-types */
import React from 'react';

function KitchenOrders({ orden, doneOrder }) {
  const { userId, id, resume, status, timeIn } = orden;
  return (
    <>
      <div>
        <li>Orden: {id}</li>
        <li>{timeIn}</li>
        <li>Mesero: {userId}</li>
        {resume.map((food) => (
          <section key={food.id}>
            <p>{food.name}</p>
            <p>x{food.qty}</p>
          </section>
        ))}
        <button type="button" onClick={() => doneOrder(id)}>
          {status}
        </button>
      </div>
    </>
  );
}

export default KitchenOrders;
