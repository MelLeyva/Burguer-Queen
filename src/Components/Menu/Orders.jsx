/* eslint-disable no-console */
import PropTypes from 'prop-types';
import React from 'react';
import '../../Styles-scss/Order.scss';

function Orders({ user, logOut }) {
  console.log(user);
  return (
    <div className="breakfast">
      <h1 className="title">Aqui van los desayunos </h1>
      <button
        className="exit"
        type="button"
        onClick={(e) => {
          e.preventDefault();
          logOut();
          console.log('ya salÃ­');
        }}
      >
        Salir
      </button>
    </div>
  );
}

Orders.propTypes = {
  user: PropTypes.string.isRequired,
  logOut: PropTypes.func.isRequired
};

export default Orders;
