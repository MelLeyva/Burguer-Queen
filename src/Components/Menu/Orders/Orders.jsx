import React from 'react';

function Orders() {
  return <div>Estas son las ordenes.</div>;
}

export default Orders;
