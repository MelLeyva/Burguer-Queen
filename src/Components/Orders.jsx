/* eslint-disable no-console */
import React from 'react';
import { useNavigate } from 'react-router-dom';
// import { Link } from 'react-router-dom';
import { signOut } from 'firebase/auth';
import { auth } from '../firebase-config';
import '../Styles-scss/Order.scss';

function BreakFast(usuario) {
  const navigate = useNavigate();
  console.log(usuario);

  const logout = async () => {
    await signOut(auth);
    navigate('/');
  };

  return (
    <div className="breakfast">
      <h1 className="title">Aqui van los desayunos </h1>
      <button className="exit" type="button" onClick={logout}>
        Salir
      </button>
    </div>
  );
}

export default BreakFast;
