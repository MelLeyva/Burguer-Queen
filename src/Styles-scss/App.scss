body {
    background-image: url('../img/background.jpg');
    background-size: 100% 100%;
    height: 1024px;
}
* {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }