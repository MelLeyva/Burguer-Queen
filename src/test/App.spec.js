/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import '@testing-library/jest-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import App from '../App';
import Orders from '../Components/Menu/Orders';

test('About App component', () => {
  render(<App />);

  const contentEmail = screen.getByPlaceholderText('Correo electrónico');
  const contentPassword = screen.getByPlaceholderText('Contraseña');
  const contentButton = screen.getByRole('button', { name: 'Ingresar' });

  expect(contentEmail).toBeInTheDocument();
  expect(contentPassword).toBeInTheDocument();
  expect(contentButton).toBeInTheDocument();

  fireEvent.click(contentButton);
  expect(render(<Orders />)).toMatchSnapshot();

  const titleByText = screen.getByText('Aqui van los desayunos');
  expect(titleByText).toBeInTheDocument();
});
