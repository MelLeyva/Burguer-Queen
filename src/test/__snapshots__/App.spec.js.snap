// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <div
            class="chalkboard"
          >
            <div
              class="burger-queen"
            >
              <img
                alt="crown"
                class="crown"
                src="crown.png"
              />
              ;
              <img
                alt="hamburger"
                class="burger"
                src="hamburger.png"
              />
              ;
            </div>
            <div
              class="form"
            >
              <h1
                class="title"
              >
                BURGER QUEEN
              </h1>
              <h2
                class="log.in"
              >
                Iniciar sesión
              </h2>
              <form
                class="inputs"
              >
                <select>
                  <option>
                    Administrador
                  </option>
                  <option>
                    Mesero
                  </option>
                  <option>
                    Cocina
                  </option>
                </select>
                <input
                  class="e-mail"
                  placeholder="Correo electrónico"
                  value=""
                />
                <input
                  class="password"
                  placeholder="Contraseña"
                  type="password"
                  value=""
                />
                <button
                  type="submit"
                >
                  Ingresar
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="breakfast"
      >
        <h1
          class="title"
        >
          Aqui van los desayunos 
        </h1>
        <button
          class="exit"
          type="button"
        >
          Salir
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="breakfast"
    >
      <h1
        class="title"
      >
        Aqui van los desayunos 
      </h1>
      <button
        class="exit"
        type="button"
      >
        Salir
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
